modules/core/keeper/keeper.go

```diff
@@ -64,14 +64,14 @@ func NewKeeper(
 	clientKeeper := clientkeeper.NewKeeper(cdc, key, paramSpace, stakingKeeper, upgradeKeeper)
 	connectionKeeper := connectionkeeper.NewKeeper(cdc, key, paramSpace, clientKeeper)
 	portKeeper := portkeeper.NewKeeper(scopedKeeper)
-	channelKeeper := channelkeeper.NewKeeper(cdc, key, clientKeeper, connectionKeeper, portKeeper, scopedKeeper)
+	channelKeeper := channelkeeper.NewKeeper(cdc, key, clientKeeper, connectionKeeper, &portKeeper, scopedKeeper)
 
 	return &Keeper{
 		cdc:              cdc,
 		ClientKeeper:     clientKeeper,
 		ConnectionKeeper: connectionKeeper,
 		ChannelKeeper:    channelKeeper,
-		PortKeeper:       portKeeper,
+		PortKeeper:       &portKeeper,
```

Comment:
the port keeper's router is assigned [here](https://github.com/cosmos/ibc-go/blob/e093d85b533ab3572b32a7de60b88a0816bed4af/modules/core/keeper/keeper.go#L90) however k.portKeeper is a value type instead of a reference type, so the assignment is not preserved across function calls. This means that if `k.portKeeper.Router` is ever accessed, it will have been an empty value.

Reply:
nice catch!

Reply:
We only ever accessed the router via `channelKeeper.Router` and never through `portKeeper.Router`